### ISSUE TITLE
RuntimeError: Error in dlopen: libtorch_cuda_linalg.so: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by zhaozhiming37** [triton-inference-server/server#4934]
**Description**

It seems cannot found `libtorch_cuda_linalg.so`

server
```
I0929 01:52:22.008062 5902 grpc_server.cc:3587] New request handler for ModelInferHandler, 0
I0929 01:52:22.008083 5902 infer_request.cc:713] [request id: 1-1] prepared: [0x0x7fd828016730] request id: 1-1, model: face_iqa, requested version: 1, actual version: 1, flags: 0x0, correlation id: 0, batch size: 0, priority: 0, timeout (us): 0
original inputs:
[0x0x7fd828008058] input: LANDMARKS, type: FP32, original shape: [2,5,2], batch + shape: [2,5,2], shape: [2,5,2]
[0x0x55929f3383b8] input: IMAGE, type: UINT8, original shape: [1,720,1280,3], batch + shape: [1,720,1280,3], shape: [1,720,1280,3]
override inputs:
inputs:
[0x0x55929f3383b8] input: IMAGE, type: UINT8, original shape: [1,720,1280,3], batch + shape: [1,720,1280,3], shape: [1,720,1280,3]
[0x0x7fd828008058] input: LANDMARKS, type: FP32, original shape: [2,5,2], batch + shape: [2,5,2], shape: [2,5,2]
original requested outputs:
SCORES
requested outputs:
SCORES

I0929 01:52:22.008134 5902 infer_request.cc:713] [request id: 1-1] prepared: [0x0x7fd828018b50] request id: 1-1, model: face_iqa_preprocess, requested version: -1, actual version: 1, flags: 0x0, correlation id: 0, batch size: 0, priority: 0, timeout (us): 0
original inputs:
[0x0x7fd828019018] input: landmarks__1, type: FP32, original shape: [2,5,2], batch + shape: [2,5,2], shape: [2,5,2]
[0x0x7fd828018e88] input: image__0, type: UINT8, original shape: [1,720,1280,3], batch + shape: [1,720,1280,3], shape: [1,720,1280,3]
override inputs:
inputs:
[0x0x7fd828018e88] input: image__0, type: UINT8, original shape: [1,720,1280,3], batch + shape: [1,720,1280,3], shape: [1,720,1280,3]
[0x0x7fd828019018] input: landmarks__1, type: FP32, original shape: [2,5,2], batch + shape: [2,5,2], shape: [2,5,2]
original requested outputs:
crops__0
requested outputs:
crops__0

I0929 01:52:22.008203 5902 libtorch.cc:2142] model face_iqa_preprocess, instance face_iqa_preprocess_0_1, executing 1 requests
I0929 01:52:22.008216 5902 libtorch.cc:992] TRITONBACKEND_ModelExecute: Running face_iqa_preprocess_0_1 with 1 requests
I0929 01:52:22.008273 5902 pinned_memory_manager.cc:161] pinned memory allocation: size 2764800, addr 0x7fd9d2000090
I0929 01:52:22.008507 5902 pinned_memory_manager.cc:161] pinned memory allocation: size 80, addr 0x7fd9d22a30a0
I0929 01:52:22.033392 5902 grpc_server.cc:3758] ModelInferHandler::InferResponseComplete, 0 step ISSUED
I0929 01:52:22.033527 5902 grpc_server.cc:3312] ModelInferHandler::InferRequestComplete
I0929 01:52:22.033538 5902 grpc_server.cc:3594] Process for ModelInferHandler, rpc_ok=1, 0 step COMPLETE
I0929 01:52:22.033545 5902 pinned_memory_manager.cc:190] pinned memory deallocation: addr 0x7fd9d2000090
I0929 01:52:22.033550 5902 grpc_server.cc:2504] Done for ModelInferHandler, 0
I0929 01:52:22.033554 5902 pinned_memory_manager.cc:190] pinned memory deallocation: addr 0x7fd9d22a30a0
```
client
```
Traceback (most recent call last):
  File "/media/zzm/Code/work/cv_face_quality/face_iqa.py", line 144, in <module>
    FaceIqa.request(
  File "/media/zzm/Code/work/cv_face_quality/face_iqa.py", line 118, in request
    raise e
  File "/media/zzm/Code/work/cv_face_quality/face_iqa.py", line 109, in request
    response = client.infer(
  File "/home/zzm/anaconda3/envs/triton/lib/python3.8/site-packages/tritonclient/grpc/__init__.py", line 1322, in infer
    raise_error_grpc(rpc_error)
  File "/home/zzm/anaconda3/envs/triton/lib/python3.8/site-packages/tritonclient/grpc/__init__.py", line 62, in raise_error_grpc
    raise get_error_grpc(rpc_error) from None
tritonclient.utils.InferenceServerException: [StatusCode.INTERNAL] in ensemble 'face_iqa', PyTorch execute failure: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript, serialized code (most recent call last):
  File "code/__torch__.py", line 12, in forward
    landmarks: Tensor) -> Tensor:
    face_align = self.face_align
    crops = (face_align).forward(img, landmarks, )
             ~~~~~~~~~~~~~~~~~~~ <--- HERE
    norm = self.norm
    return (norm).forward(crops, )
  File "code/__torch__.py", line 29, in forward
    _1 = __torch__.torch.nn.functional.grid_sample
    src_shape = torch.slice(torch.size(img), 1, 3)
    theta = (self).get_similarity_transform(src_pts, src_shape, )
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    _2 = torch.permute(img, [0, 3, 1, 2])
    _3 = [(torch.size(theta))[0], (torch.size(img))[3], (torch.size(img))[1], (torch.size(img))[2]]
  File "code/__torch__.py", line 44, in get_similarity_transform
    src = torch.pad(src_pts, [0, 1], "constant", 1.)
    ref_pts = self.ref_pts
    _7, _8, _9, _10 = torch.linalg_lstsq(ref_pts, src)
                      ~~~~~~~~~~~~~~~~~~ <--- HERE
    tfm = torch.transpose(_7, 1, 2)
    h1 = src_shape[0]

Traceback of TorchScript, original code (most recent call last):
  File "/media/zzm/Code/work/cv-face-image-quality-assessment/fiqa_exp/ser_exp/pipeline.py", line 188, in forward
            faces: torch Tensor
        """
        crops = self.face_align(img, landmarks)
                ~~~~~~~~~~~~~~~ <--- HERE
        crops = self.norm(crops)
        return crops
  File "/media/zzm/Code/work/cv-face-image-quality-assessment/fiqa_exp/ser_exp/pipeline.py", line 108, in forward
    def forward(self, img, src_pts):
        src_shape = img.shape[1:3]
        theta = self.get_similarity_transform(src_pts, src_shape)
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        # print(f"img shape: {img.shape}")
        imgs = img.permute(0, 3, 1, 2).\
  File "/media/zzm/Code/work/cv-face-image-quality-assessment/fiqa_exp/ser_exp/pipeline.py", line 95, in get_similarity_transform
    def get_similarity_transform(self, src_pts: torch.Tensor, src_shape: List[int]):
        src = F.pad(src_pts, (0, 1), "constant", 1.0)
        res = torch.linalg.lstsq(self.ref_pts, src)
              ~~~~~~~~~~~~~~~~~~ <--- HERE
        tfm = res.solution.transpose(1, 2)
        h1, w1 = src_shape[0], src_shape[1]
RuntimeError: Error in dlopen: libtorch_cuda_linalg.so: cannot open shared object file: No such file or directory
```

**Triton Information**

triton docker version
`nvcr.io/nvidia/tritonserver:22.08-py3`

**To Reproduce**

**Expected behavior**
no error

**Comment by sau00** [comment]
Facing the same problem. Any updates?

**Comment by nv-kmcgill53** [comment]
Thank you for reporting this. We have created a ticket for this issue.

**Comment by zhaozhiming37** [comment]
> Thank you for reporting this. We have created a ticket for this issue.

Is there any update?

**Comment by ZZM37** [comment]
@nv-kmcgill53 Is there any update for this issue?

**Comment by sau00** [comment]
Any updates?

**Comment by nv-kmcgill53** [comment]
No update as of now.

**Comment by nv-kmcgill53** [comment]
Hi @zhaozhiming37, could you please provide steps to reproduce this issue? Please include as well the entire command line which is used to start Triton. This will help in our investigation. Thanks.

**Comment by zhaozhiming37** [comment]
## torchscript model
```
import torch
import torch.nn.functional as F
from torch.nn import Module


class LSTSQ(Module):
    def __init__(self, ref_pts):
        super(LSTSQ, self).__init__()
        ref_pts = torch.as_tensor(ref_pts)
        self.register_buffer("ref_pts", ref_pts)

    def forward(self, src: torch.Tensor):
        src = F.pad(src, (0, 1), "constant", 1.0)
        res = torch.linalg.lstsq(self.ref_pts, src)
        tfm = res.solution
        return tfm


if __name__ == '__main__':
    ref_pts = [[54.70656938, 73.85185552],
               [105.04542462, 73.57342622],
               [80.03599766, 102.48085786],
               [59.35614298, 131.95071209],
               [101.04271195, 131.7201409]]
    lstsq = LSTSQ(ref_pts).eval()
    script = torch.jit.script(lstsq)
    script.save("./model.pt")
```
## triton server configure
```
platform: "pytorch_libtorch"
input {
  name: "points__0"
  data_type: TYPE_FP32
  dims: 5
  dims: 2
}
output {
  name: "tfm__0"
  data_type: TYPE_FP32
  dims: 2
  dims: 3
}
instance_group {
  count: 1
  kind: KIND_GPU
}
model_warmup {
  name: "zero_tensor"
  inputs {
    key: "points__0"
    value {
      data_type: TYPE_FP32
      dims: 5
      dims: 2
      zero_data: true
    }
  }
}
model_warmup {
  name: "random_tensor"
  inputs {
    key: "points__0"
    value {
      data_type: TYPE_FP32
      dims: 5
      dims: 2
      random_data: true
    }
  }
}
```
## triton server command line
`tritonserver --model-repository /models --log-verbose 1`
## client test
```
if __name__ == "__main__":
    import numpy as np
    import tritonclient.grpc as grpcclient
    from tritonclient.utils import np_to_triton_dtype

    data = np.asarray([[54.70656938, 73.85185552],
                       [105.04542462, 73.57342622],
                       [80.03599766, 102.48085786],
                       [59.35614298, 131.95071209],
                       [101.04271195, 131.7201409]], dtype=np.float32)
    inputs = [
        grpcclient.InferInput(
            "points__0",
            data.shape,
            np_to_triton_dtype(data.dtype)
        ),
    ]
    inputs[0].set_data_from_numpy(data)
    client = grpcclient.InferenceServerClient("127.0.0.1:8001")
    client.infer("lstsq", inputs)
```
## issue scription
When triton server start, there is no exception. However, when I request the service, I get the issue.

**Comment by zhaozhiming37** [comment]
@nv-kmcgill53 any updates or plans?

**Comment by nv-kmcgill53** [comment]
This issue has not been scheduled as of yet.

**Comment by karpp** [comment]
Also faced this issue on docker version `nvcr.io/nvidia/tritonserver:22.11-py3`. Are there any updates?

**Comment by dyastremsky** [comment]
Thank you for submitting this issue.

This library will now be available in the libtorch backend. It is on the main branch and will be available in the upcoming 23.02 Triton release.
